agent/indy_catalyst_agent/messaging/credentials/manager.py

```diff
@@ -274,9 +327,32 @@ def context(self) -> RequestContext:
         """
         credential = json.loads(credential_message.issue)
 
-        credential_exchange_record = await CredentialExchange.retrieve_by_tag_filter(
-            self.context, tag_filter={"thread_id": credential_message._thread_id}
-        )
+        try:
+            (
+                credential_exchange_record
+            ) = await CredentialExchange.retrieve_by_tag_filter(
+                self.context, tag_filter={"thread_id": credential_message._thread_id}
+            )
+        except StorageNotFoundError:
+
+            if not credential_message._thread.pthid:
+                raise
+
+            # If the thread_id does not return any results, we check the
```

Comment:
and here

Reply:
`credential_message._thread` may be None given a naive client